rubi/rubi/client.py

```diff
@@ -70,231 +70,224 @@ def __init__(
-        )
-        self.router = RubiconRouter.from_network(
-            network=self.network, wallet=self.wallet, key=self.key
-        )
-
-        # TODO a Dict of { symbol: ERC20 }, investigate
-        self.tokens = self.get_network_tokens()
-        self._pairs: Dict[str, Pair] = {}
-
-        self.message_queue = message_queue  # type: Queue | None
-
-        self.market_data = MarketData.from_network_with_tokens(
-            network=self.network, network_tokens=self.tokens
-        )
-
-        # Order tracking
-        self.active_limit_orders: Dict[int, ActiveLimitOrder] = {}
+        self._key = key  # type: str |  None
 
     @classmethod
     def from_http_node_url(
         cls,
         http_node_url: str,
-        custom_token_addresses_file: Optional[str] = None,
         message_queue: Optional[Queue] = None,
         wallet: Optional[Union[ChecksumAddress, str]] = None,
         key: Optional[str] = None,
+        custom_token_addresses_file: Optional[str] = None,
+        with_subgraph: bool = True,
+        **kwargs,
     ):
         """Initialize a Client using a http_node_url.
 
         :param http_node_url: URL of the HTTP node.
         :type http_node_url: str
-        :param custom_token_addresses_file: The name of a yaml file (relative to the current working directory) with
-            custom token addresses. Overwrites the token config found in network_config/{chain}/network.yaml.
-            (optional, default is None).
-        :type custom_token_addresses_file: Optional[str]
         :param message_queue: Optional message queue for processing events (optional, default is None).
         :type message_queue: Optional[Queue]
         :param wallet: Wallet address (optional, default is None).
         :type wallet: Optional[Union[ChecksumAddress, str]]
         :param key: Key for the wallet (optional, default is None).
         :type key: str
+        :param custom_token_addresses_file: The name of a yaml file (relative to the current working directory) with
+            custom token addresses. Overwrites the token config found in network_config/{chain}/network.yaml.
+            (optional, default is None).
+        :type custom_token_addresses_file: Optional[str]
+        :param with_subgraph: Should the Network be instantiated with market data from the subgraph.
+        :type with_subgraph: bool
         """
-        network = Network.from_config(
+        network = Network.from_http_node_url(
             http_node_url=http_node_url,
             custom_token_addresses_file=custom_token_addresses_file,
+            with_subgraph=with_subgraph,
         )
 
-        return cls(network=network, message_queue=message_queue, wallet=wallet, key=key)
+        return cls(
+            network=network,
+            message_queue=message_queue,
+            wallet=wallet,
+            key=key,
+        )
 
     ######################################################################
-    # pair methods
```

Comment:
i completely agree that moving away from this `pair` based approach is the move. my open question here is how we should handle the deprecation of these functions. imo, instead of gutting them in this `pr`, it may be better to start throwing `DeprecationWarning` when these functions are called as to not break any existing systems when they upgrade